src/renderer/components/+workloads-overview/overview.tsx

```diff
@@ -61,9 +59,10 @@ export class WorkloadsOverview extends React.Component<Props> {
     if (isAllowedResource("events")) {
       stores.push(eventStore);
     }
-    this.isReady = stores.every(store => store.isLoaded);
-    await Promise.all(stores.map(store => store.loadAll()));
-    this.isReady = true;
+
+    for (const store of stores) {
+      await store.loadAll();
+    }
```

Comment:
This might lead to memory leaks and/or errors IMHO.
It creates sequential loading for all stores, so the code **after** for-cycle might be evaluated _in theory_ when user already has left current page (but stores still loading data). For big clusters could be real-case scenario?

Reply:
Or actually I might be wrong.. since unsubscribing would happen anyway:
```ts
await when(() => this.isUnmounting);
```

Reply:
But still for big clusters could be a traffic issue since it's continue to load all stores even if user left the page.

Reply:
Probably not worse than the current implementation which triggers all requests in parallel?

Reply:
Yeah, for sure